src/components/Buttons/Button.js

```diff
@@ -0,0 +1,290 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Radium from 'radium'
+import Icon from '../Icon/Icon'
+import withTheme from '../../styles/themer/withTheme'
+import { themePropTypes } from '../../styles/themer/utils'
+import { spacing, colors } from '../../styles'
+import tinycolor from 'tinycolor2'
+
+const noop = () => {} // eslint-disable-line no-empty-function
+
+const baseStyles = {
+  touchAction: 'manipulation',
+  cursor: 'pointer',
+  border: '1px solid transparent',
+  borderRadius: '4px',
+  fontWeight: 600,
+  whiteSpace: 'nowrap',
+  userSelect: 'none',
+  WebkitFontSmoothing: 'antialiased',
+
+  // Reset unusual Firefox-on-Android default style;
+  // see https://github.com/necolas/normalize.css/issues/214
+  backgroundImage: 'none',
+
+  // Takes care of the heavier default font-weights in firefox
+  MozOsxFontSmoothing: 'grayscale',
+
+  // Ensures any icons stay vertically centered
+  display: 'inline-flex',
+  alignItems: 'center',
+
+  ':focus': {
+    outline: 'none'
+  }
+}
+
+const sizeStyles = {
+  tiny: {
+    ...spacing.PADDING_X_XS,
+    height: '24px',
+    fontSize: '11px',
+    textTransform: 'uppercase'
+  },
+
+  small: {
+    ...spacing.PADDING_X_XS,
+    fontSize: '14px',
+    height: '32px'
+  },
+
+  standard: {
+    ...spacing.PADDING_X_SM,
+    fontSize: '16px',
+    height: '40px'
+  },
+
+  large: {
+    ...spacing.PADDING_X_MD,
+    fontSize: '18px',
+    height: '48px'
+  }
+}
+
+const linkStyles = {
+  display: 'flex',
+  alignItems: 'center',
+  boxSizing: 'border-box'
+}
+
+const getSnacksStyles = props => {
+  const { action, actionHover, primaryBackground } = props.snacksTheme.colors
+  let actionActive
+  try {
+    actionActive = tinycolor(actionHover).darken(5).toHexString()
```

Comment:
@dcocchia Giving a small lib a try and it feels much better, it was a bigger investment than I expected to try and put together a reliable `darker` function. This isn't exact as far as the color math of the design system, so I can double check with Jordan and see what we can do. If we want to take the same color steps (eg. from green 400 => 500 => 600) for resting => hover => action, I'll need to be more precise and deal with hsl colors.

Another issue - the default snacks theme has the wrong hover color, it's green 700 when it should be green 600 (with the active state being green 700). Can figure that out tomorrow as well.

Reply:
Interesting that it's wrapped in a try/catch. Is there a high chance of this failing?